job/replay_syncer.go

```diff
@@ -0,0 +1,190 @@
+	if err != nil {
+		return err
+	}
+
+	return updateCompletedReplays(stateSummary, replaySpecRepo, replayRequest.ID)
+}
+
+func (s Syncer) prepareJobSpecMap(projSpec models.ProjectSpec) (map[string]models.JobSpec, error) {
+	projectJobSpecRepo := s.projectJobSpecRepoFactory.New(projSpec)
+	jobSpecs, err := projectJobSpecRepo.GetAll()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve jobs")
+	}
+
+	for i, jSpec := range jobSpecs {
+		if jobSpecs[i].Assets, err = s.assetCompiler(jSpec, s.Now()); err != nil {
+			return nil, errors.Wrap(err, "asset compilation")
+		}
+	}
+
+	var resolvedSpecs []models.JobSpec
+	for _, jobSpec := range jobSpecs {
+		resolvedSpec, err := s.dependencyResolver.Resolve(projSpec, projectJobSpecRepo, jobSpec, nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to resolve dependency for %s", jobSpec.Name)
+		}
+		resolvedSpecs = append(resolvedSpecs, resolvedSpec)
+	}
+	jobSpecMap := make(map[string]models.JobSpec)
+	for _, currSpec := range jobSpecs {
+		jobSpecMap[currSpec.Name] = currSpec
+	}
+	return jobSpecMap, nil
+}
+
+func (s Syncer) checkInstanceState(ctx context.Context, projSpec models.ProjectSpec, allNodes []*tree.TreeNode,
+	startDate time.Time, endDate time.Time) (map[string]int, error) {
+	stateSummary := make(map[string]int)
+	stateSummary[models.InstanceStateRunning] = 0
+	stateSummary[models.InstanceStateFailed] = 0
+	stateSummary[models.InstanceStateSuccess] = 0
+
+	for _, node := range allNodes {
+		batchEndDate := endDate.AddDate(0, 0, 1).Add(time.Second * -1)
+		jobStatusAllRuns, err := s.scheduler.GetDagRunStatus(ctx, projSpec, node.GetName(), startDate, batchEndDate, schedulerBatchSize)
+		if err != nil {
+			return nil, err
+		}
+		for _, jobStatus := range jobStatusAllRuns {
+			stateSummary[jobStatus.State.String()] = stateSummary[jobStatus.State.String()] + 1
+		}
+	}
+	return stateSummary, nil
+}
+
+func updateCompletedReplays(stateSummary map[string]int, replaySpecRepo store.ReplaySpecRepository, replayID uuid.UUID) error {
+	if stateSummary[models.InstanceStateRunning] == 0 && stateSummary[models.InstanceStateFailed] > 0 {
+		if updateStatusErr := replaySpecRepo.UpdateStatus(replayID, models.ReplayStatusFailed, models.ReplayMessage{
+			Type:    models.ReplayStatusFailed,
+			Message: ReplayMessageFailed,
+		}); updateStatusErr != nil {
+			logger.I(fmt.Sprintf("marking replay as failed error: %s", updateStatusErr))
+		}
+	} else if stateSummary[models.InstanceStateRunning] == 0 && stateSummary[models.InstanceStateFailed] == 0 && stateSummary[models.InstanceStateSuccess] > 0 {
+		if updateStatusErr := replaySpecRepo.UpdateStatus(replayID, models.ReplayStatusSuccess, models.ReplayMessage{
+			Type:    models.ReplayStatusSuccess,
+			Message: ReplayMessageSuccess,
+		}); updateStatusErr != nil {
+			logger.I(fmt.Sprintf("marking replay as success error: %s", updateStatusErr))
+		}
+	}
+	return nil
```

Comment:
I think you need to return error here `updateStatusErr `

Reply:
added the return